js/util.js

```diff
@@ -478,6 +482,67 @@
         return geolocationModule;
     })();
 
+    /******************************
+    *
+    * Sprite mini-library
+    *
+    *
+    *******************************/
+
+    function Sprite(drawable){
+        // drawable can be a shape function, an image, or text
+        // wrap image with a function, make sure all are centred on 0,0
+        this.drawable = drawable || defaultDrawable;
+        this.position = new Vector(0,0);
+        this.facing = new Vector(-PI/2,0.1);
+        this.velocity = new Vector(0,0.1);
+    }
+
+    Sprite.prototype.accelerate = function(speed){
+        this.velocity = add(this.velocity, multiply(this.facing, speed));
+        // console.log('position: %s, velocity: %s, facing: %s', strv(this.position), strv(this.velocity), strv(this.facing));
+    }
+
+    Sprite.prototype.applyForce = function(vec){
+        this.velocity = add(this.velocity, vec);
+    }
+
+    Sprite.prototype.rotate = function(r){
+        this.facing = this.facing.rotate(r);
+        console.log('position: %s, velocity: %s, facing: %s', strv(this.position), strv(this.velocity), strv(this.facing));
+    }
+
+    Sprite.prototype.move = function(){
+        this.position = add(this.position, this.velocity);
+    }
+
+    Sprite.prototype.draw = function(ctx){
+        ctx.rotate(this.facing.radians()); // drawable should be centered on 0,0
+        ctx.translate(this.position.x, this.position.y);
+        this.drawable.draw(ctx);
+        ctx.setTransform(1,0,0,1,0,0); // back to identity matrix
+    }
+
+    function defaultDrawable(ctx){
+        // ctx.save();
```

Comment:
Ah, I had that in there and was testing if it was needed. Doesn't appear to be needed, can probably just delete those lines.